packages/adapter-d1/src/d1.ts

```diff
@@ -94,23 +91,30 @@ class D1Queryable<ClientT extends StdClient> implements Queryable {
     const tag = '[js::execute_raw]'
     // console.debug(`${tag} %O`, query)
 
-    // TODO: rows_written or changes? Only rows_written is documented.
-    return (await this.performIO(query)).map(({ meta }) => meta.changes ?? 0)
+    return (await this.performIO(query)).map(({ meta }) => meta.rows_written ?? 0)
   }
 
   private async performIO(query: Query): Promise<Result<PerformIOResult>> {
-    // const release = await this[LOCK_TAG].acquire()
     // console.debug({ query })
 
     try {
-      // Hack for
+      // Hack for booleans, we must convert them to 0/1.
       // ✘ [ERROR] Error in performIO: Error: D1_TYPE_ERROR: Type 'boolean' not supported for value 'true'
       query.args = query.args.map((arg) => {
         if (arg === true) {
           return 1
         } else if (arg === false) {
           return 0
         }
+        // Temporary unblock for "D1_TYPE_ERROR: Type 'bigint' not supported for value '20'"
+        // For 0-legacy-ports.query-raw tests
+        // https://github.com/prisma/team-orm/issues/878
+        else if (typeof arg === 'bigint') {
+          return Number(arg)
```

Comment:
Can it be done as `String(arg)`? Or this was a cause of an error mentioned above.

Reply:
That's a great suggestion, I'm not sure if we tried that before, but it's what we want indeed!

Reply:
I made the change in https://github.com/prisma/prisma/pull/22750